pyscraper/committees/commons_api.py

```diff
@@ -0,0 +1,313 @@
+
+
+class Contact(CamelModel):
+    email: Optional[str] = None
+    phone: Optional[str] = None
+    address: Optional[str] = None
+    contact_disclaimer: Optional[str] = None
+
+
+class Category(CamelModel):
+    id: int
+    name: str
+
+
+class CommitteeType(CamelModel):
+    id: int
+    name: str
+    committee_category: Category
+
+
+class NameHistory(CamelModel):
+    id: int
+    committee_id: int
+    name: str
+    start_date: datetime
+    end_date: Optional[datetime] = None
+
+
+class Department(CamelModel):
+    department_id: int
+    name: str
+
+
+class LeadHouse(CamelModel):
+    is_commons: bool
+    is_lords: bool
+
+
+class Committee(CamelModel):
+    id: int
+    name: str
+    house: str
+    purpose: Optional[str] = None
+    contact: Optional[Contact] = None
+    parent_committee: Optional[Committee] = None
+    sub_committees: Optional[List[Committee]] = None
+    lead_house: Optional[LeadHouse] = None
+    category: Optional[Category] = None
+    committee_types: List[CommitteeType]
+    show_on_website: bool
+    website_legacy_url: Optional[str] = None
+    website_legacy_redirect_enabled: bool
+    start_date: datetime
+    end_date: Optional[datetime] = None
+    date_commons_appointed: Optional[datetime] = None
+    date_lords_appointed: Optional[datetime] = None
+    scrutinising_departments: List[Department] = []
+    is_lead_committee: Optional[bool] = None
+    name_history: List[NameHistory]
+    members: List[Member] = Field(default_factory=list)
+
+    def parl_url(self):
+        return f"https://committees.parliament.uk/committee/{self.id}/"
+
+    def expand(self):
+        """
+        Get the full committee information from the API
+        """
+        url = f"https://committees-api.parliament.uk/api/Committees/{self.id}?includeBanners=false&showOnWebsiteOnly=false"
+        response = httpx.get(url, timeout=30)
+        response.raise_for_status()
+        data = response.json()
+        return Committee.model_validate(data).add_members()
+
+    def add_members(self):
+        """
+        Fetch and add members to the committee.
+        """
+
+        def one_request(skip=0):
+            url = f"https://committees-api.parliament.uk/api/Committees/{self.id}/Members?MembershipStatus=Current&ShowOnWebsiteOnly=true&skip={skip}"
+            response = httpx.get(url, timeout=30)
+            response.raise_for_status()
+            return response.json()
+
+        data = committee_app_loop(one_request)
+        self.members = [Member.model_validate(item) for item in data]
+        return self
+
+
+class CommitteeList(RootModel[list[Committee]]):
+    def expand(self):
+        new_items = []
+
+        for item in tqdm(self.root, disable=not VerboseSettings.verbose):
+            expanded = item.expand()
+            new_items.append(expanded)
+
+        return CommitteeList(root=new_items)
+
+    def __iter__(self):
+        return iter(self.root)
+
+
+class ApiCallWithSkip(Protocol):
```

Comment:
this is just me being picky but it'd be nice if this and the next were declared further up the file so that if you are reading through it then you see these before you see the first use. Also, possibly a quick "this is what this is" comment.